content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,235 @@
+
+Our API endpoints could be:
+
+* `/rooms` to return all chat rooms,
+* `/rooms/:id/` to return a particular chat room identified by an ID,
+* `/rooms/:id/messages` to return all messages from a particular room.
+
+To post a message, our front-end will do a POST request to `/rooms/:id/messages`.
+
+We will store each message like this:
+
+    {
+        "message": "I know nothing",
+        "last_modified": 1480340246147,
+        "author": "Jon Snow",
+        "id": "a1e6e879-5c94-4a01-9ec2-28e152d1d8f4"
+    }
+
+
+The JavaScript code could use whatever library or latest hype JS framework but in our example we will stick with modern [Vanilla JS](http://vanilla-js.com/) and use top notch [ES2015 syntax](https://babeljs.io/docs/learn-es2015/).
+
+## API creation
+
+Hmm… what backend should we use? Django? Express? Firebase?… oh wait, why don’t use [Kinto](http://www.kinto-storage.org/)?
+
+Kinto is a perfect fit for this application as once installed, Kinto provides a flexible enough API to to create and handle other APIs.
+
+Let's see how to use Kinto for our instant chat WebApp.
+
+To show some pride, we will use an explicit subdomain for our main URL `https://kinto.discute.re/v1` to show our configuration rather than an agnostic https://api.discute.re/v1.
+
+Here are the different routes:
+
+- `/rooms/4815162342` becomes `/collections/4815162342` ;
+- `/rooms/4815162342/messages` becomes `/collections/4815162342/records` ;
+
+As we have seen before, we can add a new post to our backend with a simple POST request
+`/collections/4815162342/records`.
+
+A Kinto server can handle many Web applications through the concept of [*buckets*](https://kinto.readthedocs.io/en/latest/concepts.html#buckets-collections-and-records); a bucket can contain many *collections*, and a collection can contain many *records*, and records can contain arbitrary JSON data.
+
+![](https://kinto.readthedocs.io/en/latest/_images/concepts-general.png)
+{.image}
+
+For our application, we’ll create a `chat` bucket, which will contain as many collections as we want distinct chat rooms. Each room collection will itself represent chat messages through records.
+
+You can use [kinto-admin](https://github.com/Kinto/kinto-admin) plugin to easily manage your buckets, collections and records.
+
+![](https://d2mxuefqeaa7sj.cloudfront.net/s_4340C716162DF03B2B97575BFA438C542185694D96D86DB6EB44B989957A3ADE_1480537208814_file.png)
+{.image}
+
+So the request that will allow retrieving all messages from our example  `4815162342` room is:
+`GET https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records`, and the request to post a new message is:
+`POST https://kinto.discute.re/buckets/chat/collections/4815162342/records`.
+
+It's very similar to our first envisioned API calls who were
```

Comment:
`that` or `which`, not `who`, for non-person things. I think `which` is more appropriate here, and probably a comma before `which`.